### PR TITLE
Skeleton of K tutorial

### PR DESCRIPTION
**Pull request by dwightguth**
This just moves some files around and adds the skeleton of the K tutorial in terms of its file structure, readmes, and lesson organization. I intend to continue in future PRs by writing one lesson per PR.

### CODE REVIEW

**Comment by dwightguth** [comment]
This PR is failing due to now-incorrect relative paths in the k-exercises repo. I will make a PR to fix this once the other k-exercises PR is merged. I'd rather not make them at the same time to avoid potential confusion.

**Review by ehildenb — APPROVED** [review]
@dwightguth please don't forget to remove the `branch: pr` on the k-exercises checkout step before merging. Also please wait for @shd101wyy 's review of the changes to the generated website, so he can double check it won't break the online website.

**Comment by shd101wyy** [comment]
@ehildenb I just built the website on branch `k-tutorial` and it works fine :+1:

**Pull request merged**
